File: backend/football_world_cup/src/main/java/rs/ac/bg/fon/football_world_cup/domain/Utakmica.java
package rs.ac.bg.fon.football_world_cup.domain;

import jakarta.persistence.*;
import lombok.*;

@Entity

@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@Setter
@EqualsAndHashCode
@ToString

public class Utakmica {

    @Id
    @GeneratedValue
    private Long id;

    @ManyToOne
    @JoinColumn(name = "domacin_id")
//    @Valid
//    @NotNull(message = "Mora se proslediti koja reprezentacija je domacin na utakmici")
    private Reprezentacija domacin;

    @ManyToOne
    @JoinColumn(name = "gost_id")
//    @Valid
//    @NotNull(message = "Mora se proslediti koja reprezentacija je gost na utakmici")
    private Reprezentacija gost;

    @OneToOne
    @JoinColumn(name = "evidencija_utakmice_id")
    private EvidencijaUtakmice evidencijaUtakmice;

    @ManyToOne
    @JoinColumn(name = "stadion_id")
//    @Valid
//    @NotNull(message = "Mora se izabrati stadion na kojem ce se igrati utakmica")
    private Stadion stadion;

    @Embedded
//    @Valid
//    @NotNull(message = "Mora se proslediti zeljeni termin")
    private Termin termin;

}

File: frontend/src/domain/Utakmica.ts
import {Reprezentacija} from "./Reprezentacija";
import {Stadion} from "./Stadion";
import {EvidencijaUtakmice} from "./EvidencijaUtakmice";

export class Utakmica {
    public id?: number;
    public domacin: Reprezentacija;
    public gost: Reprezentacija;
    public evidencijaUtakmice?: EvidencijaUtakmice;
    public stadion: Stadion;
    public odigrana: boolean;
    public termin: {
        pocetak: Date;
        kraj: Date;
    };

    constructor(id: number, domacin: Reprezentacija, gost: Reprezentacija, evidencijaUtakmice: EvidencijaUtakmice, stadion: Stadion, odigrana:boolean, termin: {
        pocetak: Date;
        kraj: Date
    }) {
        this.id = id;
        this.domacin = domacin;
        this.gost = gost;
        this.evidencijaUtakmice = evidencijaUtakmice;
        this.stadion = stadion;
        this.odigrana = odigrana;
        this.termin = termin;
    }
}
